### PR TITLE
Render homepage & job details map

### PR DESCRIPTION
**Pull request by riya-n**
Note that this does not include the Places API (so the coordinates for the jobs are still hard-coded values).
The homepage map has a marker for each job with an info window with some details about the job. The user can double click on the marker to be taken to the job details page. The job details page also includes a map, but only with the marker of that job.

This is how it looks on the homepage:
image

### CODE REVIEW

File: walk-in-interview/src/main/webapp/script.js [hunk 1]
```diff
@@ -256,6 +261,22 @@ function buildJobElement(job) {
     detailsForm.submit();
   });
 
+  const marker = addMarker(map, job);
+  marker.addListener('click', function() {
+    new google.maps.InfoWindow({
+      content: jobListing.innerHTML,
```
Comment (SibingWu) [inline walk-in-interview/src/main/webapp/script.js]: * does the jobListing mean one job post listed on the home page? maybe rename to jobPost?
* can u take a screenshot of how it looks like? i am not sure if it is user-friendly to contain all info appearing on job post on homepage in a small info window, especially this web is viewed on mobile phone screen

File: walk-in-interview/src/main/webapp/maps.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    center: {lat: latitude, lng: longitude},
+    zoom: zoom,
+  });
+}
+
+/**
+ * Drops the markers so that the user can see them falling.
+ * The pins indicate the location of the job. You can also
+ * click on them to get some information about them.
+ * @param {google.maps.Map} map The map to add the marker to.
```
Comment (yohanlaunay) [inline walk-in-interview/src/main/webapp/maps.js:39]: super nit: newline before.

File: walk-in-interview/src/main/webapp/maps.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+// coordinates for centering the map around Singapore
+const SG_LATITUDE = 1.3521;
+const SG_LONGITUDE = 103.8198;
+const SG_MAP_ZOOM = 11;
+// appropriate zoom for the map to show the individual job
+const JOB_MAP_ZOOM = 15;
+
+/**
+ * Creates a map in the provided html div.
+ * @param {String} mapElementId The map element id.
```
Comment (yohanlaunay) [inline walk-in-interview/src/main/webapp/maps.js:16]: super nit: newline before

File: walk-in-interview/src/main/webapp/script.js (hunk 1 above)
Comment (yohanlaunay) [reply walk-in-interview/src/main/webapp/script.js]: agreed. maybe you need a job preview content.

**Review by yohanlaunay — APPROVED** [review]

File: walk-in-interview/src/main/webapp/script.js (hunk 1 above)
Comment (riya-n) [reply walk-in-interview/src/main/webapp/script.js]: I renamed it to jobPostPreview. Here's what is looks like:
image

**Review by SibingWu — APPROVED** [review]

**Pull request merged**